core/src/program.rs

```diff
@@ -312,7 +312,7 @@ impl<EC: EvalCache> Program<EC> {
     pub fn parse(&mut self) -> Result<RichTerm, Error> {
         self.vm
             .import_resolver_mut()
-            .parse(self.main_id)
+            .parse(self.main_id, InputFormat::Nickel)
```

Comment:
I wonder if those shouldn't guess the format as well, instead of `InputFormat::Nickel`. That's why it could be handy to have a `parse_auto` (the name isn't great, so feel free to use another if you find some!). However, this one can also be done in a subsequent PR, as the current code at least fixes the original issue.

Reply:
`parse_infer_format`? less concise, but more descriptive ...

Reply:
@olorin37 right, it's a reasonable suggestion and better than `parse_auto`, so go for it :slightly_smiling_face: